app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def self.update_posts_counter(user)
+    user.posts_counter = Post.where(author: user).count
+  end
```

Comment:
- [ ] Great job creating a method that updates the posts counter for a user. 👏🏻 👏🏻  However It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) -* `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method to be a `private` method 👍🏻 💯 
- 🤔  Your code might look like this 👇🏻 :


   ```ruby
     # app/models/post.rb
     after_save :update_post_counter
   
     private
   
     def update_post_counter
       author.increment!(:posts_counter)
     end
   ```

- [ ] Make sure you to use the  `author.increment!` and not  `user.increment!`

Reply:
I didn't know about this Rails feature. I implemented it right away, and the program behaves more logically now. Thanks for the explanation!!

Reply:
### 🙏🏻 🙏🏻